src/mixins/Themeable.ts

```diff
@@ -128,6 +134,67 @@ function createThemeClassesLookup(classes: ThemeClasses[]): ClassNames {
 }
 
 /**
+ * The class for the theme injector context, used to control the theme once the
+ * theme injector has been defined in the registry.
+ */
+export class ThemeInjectorContext extends Evented {
+
+	private _theme: any;
+
+	/**
+	 * @param theme optional theme to initialize the context with
+	 */
+	constructor(theme?: any) {
+		super({});
+		this._theme = theme;
+	}
+
+	/**
+	 * @param theme the theme object to set
+	 */
+	public set(theme: any) {
+		this._theme = theme;
+		this.emit({ type: 'invalidate' });
+	}
+
+	/**
+	 * Return the current theme
+	 */
+	public get theme(): any {
+		return this._theme;
+	}
+}
+
+/**
+ * Custom ThemeInjector class that listens to the invalidate event
+ * from the context to `invalidate` any widgets the have had a theme
+ * injected.
+ */
+export class ThemeInjector extends BaseInjector<ThemeInjectorContext> {
+	constructor(context: ThemeInjectorContext) {
+		super(context);
+		this.own(context.on('invalidate', this.invalidate.bind(this)));
+	}
+}
+
+/**
+ * Convience function that given a theme and an optional registry, the theme
+ * injector is defined against the registry and the theme context return.
```

Comment:
"registry and the theme context return" -> either "registry, and the theme context is returned" or "registry, returning the theme context" ?